src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2156,8 +2166,10 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             CompilationTestData testData,
             CancellationToken cancellationToken)
         {
-            bool embedPdb = options?.DebugInformationFormat == DebugInformationFormat.Embedded;
+            options = options ?? EmitOptions.Default.WithIncludePrivateMembers(metadataPEStream == null);
+            bool embedPdb = options.DebugInformationFormat == DebugInformationFormat.Embedded;
             Debug.Assert(!embedPdb || pdbStream == null);
+            Debug.Assert(metadataPEStream == null || !options.IncludePrivateMembers);
```

Comment:
"no secondary stream" implies "include private members".

There is currently no scenario where we emit ref assemblies to the primary output stream. 
We either emit a regular assembly, or a metadata-only assembly (which includes private members, that's the IDE scenario).